src/storage/mvcc/reader.rs

```diff
@@ -176,23 +180,28 @@ impl<'a> MvccReader<'a> {
 
     pub fn get(&mut self, key: &Key, mut ts: u64) -> Result<Option<Value>> {
         // Check for locks that signal concurrent writes.
-        if let Some(lock) = try!(self.load_lock(key)) {
-            if lock.ts <= ts {
-                if ts == u64::MAX && try!(key.raw()) == lock.primary {
-                    // when ts==u64::MAX(which means to get latest committed version for
-                    // primary key),and current key is the primary key, returns the latest
-                    // commit version's value
-                    ts = lock.ts - 1;
-                } else {
-                    // There is a pending lock. Client should wait or clean it.
-                    return Err(Error::KeyIsLocked {
-                        key: try!(key.raw()),
-                        primary: lock.primary,
-                        ts: lock.ts,
-                        ttl: lock.ttl,
-                    });
+        match self.isolation_level {
+            IsolationLevel::SI => {
+                if let Some(lock) = try!(self.load_lock(key)) {
+                    if lock.ts <= ts {
```

Comment:
Could we reduce the number of conditions in each `{}` (reduce the number of `{}`), may be we could define an independent function to check the lock?